disassemblers/ddisasm/ddisasmBBorRef.py

```diff
@@ -0,0 +1,305 @@
+            return RefKind.D2C
+        elif (
+            isinstance(from_block, gtirb.DataBlock)
+            and isinstance(to_block, gtirb.DataBlock)
+        ):
+            return RefKind.D2D
+        else:
+            assert False
+
+    for m in ir.modules:
+        for block in m.byte_blocks:
+            sym_exprs = block.byte_interval.symbolic_expressions_at(
+                range(block.address, block.address + block.size)
+            )
+            for bi, offset, sym_expr in sym_exprs:
+                from_addr = bi.address + offset
+
+                for symbol in sym_expr.symbols:
+                    if symbol._payload is None:
+                        continue
+
+                    if isinstance(symbol._payload, gtirb.Block):
+                        tblock = symbol._payload
+                        if isinstance(tblock, gtirb.ProxyBlock):
+                            # TODO: ??
+                            continue
+
+                        to_addr = tblock.byte_interval.address + tblock.offset
+                        kind = ref_kind(block, tblock).value
+
+                    else: # int
+                        to_addr = symbol._payload
+                        kind = 0 # TODO
+
+                    ref = refInf.ref.add()
+                    update_ref(m, ref, from_addr, to_addr, kind)
+
+                    logging.debug(
+                        "%d: From 0x%x -> 0x%x"
+                        % (kind, ref.ref_va, ref.target_va)
+                    )
+
+    logging.info("Collect Refs done! ready to write output...")
+    f = open(output, "wb")
+    f.write(refInf.SerializeToString())
+    f.close()
+
+
+class FileType(Enum):
+    IR = "ir"
+    ELF = "elf"
+    PE = "pe"
+    OTHERS = "others"
+
+
+def get_magic_file_type(file_path: Path) -> FileType:
+    """
+    Get the FileType of the given file
+    """
+    with open(file_path, "rb") as magic_fp:
+        magic = magic_fp.read(5)
+        if magic[:5] == b"GTIRB":
+            return FileType.IR
+        elif magic[:4] == b"\x7fELF":
+            return FileType.ELF
+        else:
+            dos_header = magic_fp.read(64)
+            if len(dos_header) < 64 or dos_header[:2] != b'MZ':
+                return FileType.OTHERS
+            pe_offset = int.from_bytes(dos_header[60:64], byteorder='little')
+            magic_fp.seek(pe_offset)
+            if magic_fp.read(4) == b'PE\x00\x00':
+                return FileType.PE
+            else:
+                return FileType.OTHERS
+
+
+class ExtractMode(Enum):
+    BB = "bb"
+    REF = "ref"
+
+
+if __name__ == "__main__":
+    parser = optparse.OptionParser()
+    parser.add_option(
+        "-m", "--mode",
+        metavar="string", 
+        help="Select extract-mode: `bb` or `ref`",
+        default=ExtractMode.BB.value,
+    )
+    parser.add_option(
+        "-i", "--input",
+        action="store",
+        type="string",
+        help="input binary or gtirb-IR file",
+        default=None
```

Comment:
assuming you switch to argparse I think you want
```suggestion
        type=Path,
        help="input binary or gtirb-IR file",
        required=True,
```
same for the output option

Reply:
Hmm,, I thought I would have the output as optional, but I don't really insist. So, fixed.